enhancements/olm-toggle-copied-csvs.md

```diff
@@ -0,0 +1,163 @@
+---
+title: toggle-copied-csvs
+authors:
+  - "@njhale"
+reviewers:
+  - "@TheRealJon"
+  - "@spadgett"
+  - "@bparees"
+  - "@joelanford"
+  - "@kevinrizza"
+approvers:
+  - "@bparees"
+  - "@joelanford"
+  - "@kevinrizza"
+creation-date: 2021-09-27
+last-updated: 2021-11-09
+status: implementable
+see-also:
+  - "/enhancements/olm-toggle-copied-csvs.md"
+---
+
+# Toggle Copied CSVs 
+
+## Release Signoff Checklist
+
+- [x] Enhancement is `implementable`
+- [x] Design details are appropriately documented from clear requirements
+- [ ] Test plan is defined
+- [ ] Operational readiness criteria is defined
+- [x] Graduation criteria for dev preview, tech preview, GA
+
+## Summary
+
+This proposal introduces a feature toggle that disables/enables the creation and management of [Copied CSVs](https://olm.operatorframework.io/docs/concepts/crds/operatorgroup/#copied-csvs), so that cluster admins can curb OLM's resource usage on constrained clusters.
+
+## Motivation
+
+When an Operator is installed by OLM, a stripped down copy -- e.g. a stub -- of its [CSV](https://olm.operatorframework.io/docs/concepts/crds/clusterserviceversion/) is created in every namespace the Operator is configured to watch (see [OperatorGroup](https://olm.operatorframework.io/docs/concepts/crds/operatorgroup/); collectively, these are known as _Copied CSVs_. This means that the number of Copied CSVs on a given cluster grows proportionally with both the number of namespaces and installed Operators. In the worst case, a cluster can have a Copied CSV for every Operator in every namespace, that is `|copied| = |operators|*|namespaces|`. For especially large clusters, with namespaces and installed operators tending in the hundreds or thousands, Copied CSVs consume an untenable amount of resources; e.g. OLM's memory usage, cluster Etcd limits, networking, etc. For these reasons, it's imperative that Copied CSVs be replaced with a more resource friendly successor. In the meantime, while no such successor exists, OLM should have a knob that helps curtail its high resource utilization.
+
+### Goals
+
+- Allieviate the resource utilization incurred by Copied CSVs
+- Don't block upgrades for customers of OpenShift
+- Provide an "escape hatch" that can be backported in Z-stream releases to supported minor versions of OpenShift
+
+### Non-Goals
+
+- Provide a replacement for Copied CSVs
+
+## Proposal
+
+For each non-Copied CSV, a single [Operator](https://github.com/operator-framework/enhancements/blob/master/enhancements/simplify-olm-apis.md#proposal) resource will be generated. This cluster-scoped resource will take the place of any related Copied CSVs.
+
+### User Stories
+
+#### Toggle Copied CSVs Off
+
+As a Cluster Admin, I want a way to disable Copied CSVs so that I can reduce resource consumption ahead of the feature's deprecation and eventual removal.
```

Comment:
I am not sure we are looking at feature deprecation of this particular behavior ahead of deprecation of the whole CSV API.

Reply:
That's fair -- is there some implicit ordering that the current wording implies?

Reply:
I dropped all mention deprecation here.